mono/metadata/icall.c

```diff
@@ -5809,6 +5810,54 @@ ves_icall_Mono_Runtime_EnableMicrosoftTelemetry (char *appBundleID, char *appSig
 #endif
 }
 
+ICALL_EXPORT MonoStringHandle
+ves_icall_Mono_Runtime_ExceptionToState (MonoExceptionHandle exc_handle, unsigned long *portable_hash_out, unsigned long *unportable_hash_out, MonoError *error)
+{
+#ifndef DISABLE_CRASH_REPORTING
+	// FIXME: Push handles down into mini/mini-exceptions.c
+	MonoException *exc = MONO_HANDLE_RAW (exc_handle);
+	MonoThreadSummary out;
+	mono_get_eh_callbacks ()->mono_summarize_exception (exc, &out);
+
+	*portable_hash_out = (intptr_t) GINT_TO_POINTER (out.hashes.offset_free_hash);
+	*unportable_hash_out = (intptr_t) GINT_TO_POINTER (out.hashes.offset_rich_hash);
+
+	JsonWriter writer;
+	mono_json_writer_init (&writer);
+	mono_native_state_init (&writer);
+	gboolean first_thread_added = TRUE;
+	mono_native_state_add_thread (&writer, &out, NULL, first_thread_added);
+	char *output = mono_native_state_free (&writer, FALSE);
+	return mono_string_new_handle (mono_domain_get (), output, error);
+#else
+	// Icall has platform check in managed too.
+	g_assert_not_reached ();
+#endif
+}
+
+ICALL_EXPORT void
+ves_icall_Mono_Runtime_SendMicrosoftTelemetry (char *payload, unsigned long portable_hash, unsigned long unportable_hash, MonoError *error)
+{
+#ifdef TARGET_OSX
+	if (!mono_merp_enabled ())
+		g_error ("Cannot send telemetry without registering parameters first");
+
+	pid_t crashed_pid = getpid ();
+
+	MonoStackHash hashes;
```

Comment:
Worth a `memset (&hashes, 0, sizeof (hashes))` in case it grows?

Reply:
Github doesn't show a line diff, but I fixed this.

Reply:
Would still prefer `sizeof (hashes)` though.